homeassistant/components/lg_soundbar/config_flow.py

```diff
@@ -1,51 +1,75 @@
 """Config flow to configure the LG Soundbar integration."""
-from queue import Full, Queue
+import logging
+from queue import Empty, Full, Queue
 import socket
+import uuid
 
 import temescal
 import voluptuous as vol
 
 from homeassistant import config_entries
 from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.data_entry_flow import FlowResult
 
 from .const import DEFAULT_PORT, DOMAIN
 
 DATA_SCHEMA = {
     vol.Required(CONF_HOST): str,
 }
 
+_LOGGER = logging.getLogger(__name__)
+
 
 def test_connect(host, port):
     """LG Soundbar config flow test_connect."""
     uuid_q = Queue(maxsize=1)
     name_q = Queue(maxsize=1)
 
+    def check_msg_response(response, msgs, attr):
+        msg = response["msg"]
+        if msg == msgs or msg in msgs:
+            if "data" in response and attr in response["data"]:
+                return True
+            _LOGGER.debug(
+                "[%s] msg did not contain expected attr [%s]: %s", msg, attr, response
+            )
+        return False
+
     def queue_add(attr_q, data):
         try:
             attr_q.put_nowait(data)
         except Full:
-            pass
+            _LOGGER.debug("attempted to add [%s] to full queue", data)
 
     def msg_callback(response):
-        if (
-            response["msg"] in ["MAC_INFO_DEV", "PRODUCT_INFO"]
-            and "s_uuid" in response["data"]
-        ):
+        if check_msg_response(response, ["MAC_INFO_DEV", "PRODUCT_INFO"], "s_uuid"):
             queue_add(uuid_q, response["data"]["s_uuid"])
-        if (
-            response["msg"] == "SPK_LIST_VIEW_INFO"
-            and "s_user_name" in response["data"]
-        ):
+        if check_msg_response(response, "SPK_LIST_VIEW_INFO", "s_user_name"):
             queue_add(name_q, response["data"]["s_user_name"])
 
+    sb_name = None
+    sb_uuid = None
+
     try:
         connection = temescal.temescal(host, port=port, callback=msg_callback)
         connection.get_mac_info()
         if uuid_q.empty():
             connection.get_product_info()
         connection.get_info()
-        details = {"name": name_q.get(timeout=10), "uuid": uuid_q.get(timeout=10)}
+        sb_name = name_q.get(timeout=10)
+        sb_uuid = uuid_q.get(timeout=10)
+        details = {"name": sb_name, "uuid": sb_uuid}
         return details
+    except Empty:
+        if sb_name is not None and sb_uuid is None:
+            oid = (
+                "homeassistant.components.lg_soundbar."
+                + sb_name.replace(" ", "_").lower()
+            )
+            sb_uuid = uuid.uuid5(uuid.NAMESPACE_OID, oid)
```

Comment:
I don't think this is within our unique_id requirements. The unique_id should be stable and not be able to change by user input.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler#unique-id-requirements

Reply:
What's the alternative since it looks like this type of device isn't providing any information that could be used as a UUID according to those requirements? Should it follow the `_async_handle_discovery_without_unique_id` suggestion above the requirements on that document?

Reply:
Yeah, I think that's the way to solve this.

Reply:
Although, there's no discovery here.

Reply:
If there's no uuid, I'd abort if there's already a configured config entry, otherwise we set the integration domain as unique_id.

Reply:
I don't quite understand. Isn't the integration domain required to be equal to the directory of the component? https://developers.home-assistant.io/docs/creating_integration_manifest#domain

Reply:
Yes?

Reply:
I think I'm misunderstanding this statement:

> If there's no uuid, I'd abort if there's already a configured config entry, otherwise we set the integration domain as unique_id.

I don't understand what "set the integration domain as unique_id" means. The integration domain is "lg_soundbar" for this component, or am I mixing up concepts?

Reply:
I mean that we can set the `unique_id` to the domain of the integration.

Reply:
Thanks, I've got it now. I've pushed a change for that to my branch.

Reply:
If the config entry's unique id is set to `lg_soundbar` as a fallback, it will only be possible to set up a single device from which we can't read an identifier. Isn't that an unwanted limitation?

Do you have a log from <https://github.com/home-assistant/core/issues/77524> which shows what the device sends back as a response for the requests to get the UUID?

Reply:
As suggested by @MartinHjelmare on Discord:
> We can also not set a unique id for the case when the UUID can't be read and abort on same host and use the config entry id as unique id for the entities

I think this is a better solution, it will allow multiple sound bars which don't support reading UUID 👍

Reply:
An alternative is to not set a unique id for the config entry if missing an uuid and just abort if trying to configure the same host, and use the config entry id as unique_id for the entities in that case.

We have this helper to check for same data among existing config entries of the integration and abort:
https://github.com/home-assistant/core/blob/727dcd6df6e240ec74084ffdb0baea22e3946745/homeassistant/config_entries.py#L1352-L1356

Reply:
I think that's better solution @MartinHjelmare 👍

Reply:
I've modified it to only assign a unique_id if the device returns a UUID and to abort if an integration configuration with the same host and port already exists.

https://github.com/chrismccurdy/home-assistant/blob/2d8374c0b3c2ba6bf5c465c8fd08a75a86362072/homeassistant/components/lg_soundbar/config_flow.py#L94-L99

Reply:
Good! We may want to give the entities a unique_id in the case that the config entry doesn't have a unique_id. We can use the config entry `entry_id` as entity unique_id in that case.

Reply:
How do we get the `entry_id` in the config flow? It appears that the `entry_id` isn't generated until ConfigEntriesFlowManager.async_finish_flow is called and the FlowResult returned by async_step_user is added to the config_entries.

Reply:
I'm talking about the `Entity` instances of the media_player platform. When setting up the media player the config entry will have been created. The `entry_id` is an attribute of the `ConfigEntry` instance.